src/Build/Graph/ProjectGraph.cs

```diff
@@ -73,6 +73,27 @@ public ProjectGraph(string entryProjectFile)
             string entryProjectFile,
             IDictionary<string, string> globalProperties,
             ProjectCollection projectCollection)
+            : this(entryProjectFile, globalProperties, projectCollection, false)
+        {
+        }
+
+        /// <summary>
+        /// Constructs a graph starting from the given project file, evaluating with the provided global properties and the provided project collection.
+        /// </summary>
+        /// <param name="entryProjectFile">The project file to use as the entry point in constructing the graph</param>
+        /// <param name="globalProperties">The global properties to use for all projects. May be null, in which case the containing project collection's global properties will be used.</param>
+        /// <param name="projectCollection">The collection with which all projects in the graph should be associated. May not be null.</param>
+        /// <param name="exposeProjects">
+        /// When false, only <see cref="Microsoft.Build.Execution.ProjectInstance"/> instances are exposed
+        /// from the graph. When true, the more memory and time expensive full <see cref="Microsoft.Build.Evaluation.Project"/>s
+        /// are also produced. Typically Projects are not needed for execution of a graph.
+        /// </param>
+        /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
+        public ProjectGraph(
+            string entryProjectFile,
+            IDictionary<string, string> globalProperties,
+            ProjectCollection projectCollection,
+            bool exposeProjects)
```

Comment:
We probably want this as a Flags I'd guess? Maybe something like...
```
[Flags]
public enum ProjectGraphCreationSettings
{
  CreateProjectInstance = 0,
  CreateProject = 1,
}
```

Reply:
Given comment below seems like I'll need a flag related to ImmutableWithFastItemLookup anyway.

Reply:
I don't like the useless surface area exposed by None. How about the default zero value be CreateProjectInstance, with the other flags as additional info? I'll send that in the next push.

Reply:
That's more of an Enum than a Flags then. Flags should always have a "None" (and sometimes an "All"). With enums though you'd need distinct, mutually exclusive values.

Reply:
Iterated twice on the flags. Having ProjInstance as a separate option makes no sense, as when I did that and wrote UTs to test it the basic graph generation blew up on its need for ProjectInstance. Then pulled out the PI options and changed None to Default with comments about what defaults are.